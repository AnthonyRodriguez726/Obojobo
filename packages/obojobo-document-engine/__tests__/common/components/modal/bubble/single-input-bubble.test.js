import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import SingleInputBubble from '../../../../../src/scripts/common/components/modal/bubble/single-input-bubble'

describe('SingleInputBubble', () => {
	let onClose, onCancel, onChange

	beforeEach(() => {
		onClose = jest.fn()
		onCancel = jest.fn()
		onChange = jest.fn()
	})

	test('SingleInputBubble Snapshot', () => {
		const component = renderer.create(
			<SingleInputBubble
				label="Label"
				value="Value"
				onClose={onClose}
				onCancel={onCancel}
				onChange={onChange}
			/>
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('SingleInputBubble cancels on ESC', () => {
		const el = mount(
			<SingleInputBubble
				label="Label"
				value="Value"
				onClose={onClose}
				onCancel={onCancel}
				onChange={onChange}
			/>
		)

		let input = el.find('input')

		input.simulate('keyup', {
			keyCode: 13
		})

		expect(onCancel).not.toHaveBeenCalled()

		input.simulate('keyup', {
			keyCode: 27
		})

		expect(onCancel).toHaveBeenCalled()
	})

	test('SingleInputBubble handles changes', () => {
		const el = mount(
			<SingleInputBubble
				label="Label"
				value="Value"
				onClose={onClose}
				onCancel={onCancel}
				onChange={onChange}
			/>
		)

		let input = el.find('input')

		expect(onChange).not.toHaveBeenCalled()

		input.simulate('change', {
			target: {
				value: 'Test123'
			}
		})

		expect(onChange).toHaveBeenCalledWith('Test123')
	})

	test('SingleInputBubble closes when submitted', () => {
		const el = mount(
			<SingleInputBubble
				label="Label"
				value="Value"
				onClose={onClose}
				onCancel={onCancel}
				onChange={onChange}
			/>
		)

		let form = el.find('form')

		expect(onClose).not.toHaveBeenCalled()

		form.simulate('submit')

		expect(onClose).toHaveBeenCalled()
	})
})
