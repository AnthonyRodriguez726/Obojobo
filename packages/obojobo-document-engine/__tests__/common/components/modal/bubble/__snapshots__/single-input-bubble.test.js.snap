// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleInputBubble SingleInputBubble Snapshot 1`] = `
<div
  className="obojobo-draft--components--modal--bubble"
>
  <div
    className="container"
  >
    <label
      className="single-input-bubble"
    >
      <form
        className="interactable"
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          onKeyUp={[Function]}
          type="text"
          value="Value"
        />
        <button
          onClick={[Function]}
        >
          Ok
        </button>
      </form>
      <span
        className="label"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;
